services/consensusalgo/benchmarkconsensus/test/leader_test.go

```diff
@@ -6,11 +6,121 @@ import (
 	"testing"
 )
 
-func TestLeaderProposesBlocks(t *testing.T) {
+func newLeaderHarnessAndInit(t *testing.T, ctx context.Context) *harness {
+	h := newHarness(true)
+	h.expectNewBlockProposalNotRequested()
+	h.expectCommitSent(0, h.config.NodePublicKey())
+	h.createService(ctx)
+	h.verifyCommitSent(t)
+	return h
+}
+
+func TestLeaderInit(t *testing.T) {
+	test.WithContext(func(ctx context.Context) {
+		h := newLeaderHarnessAndInit(t, ctx)
+		h.verifyNewBlockProposalNotRequested(t)
+		h.verifyHandlerRegistrations(t)
+	})
+}
+
+func TestLeaderCommitsNextBlockAfterEnoughConfirmations(t *testing.T) {
+	test.WithContext(func(ctx context.Context) {
+		h := newLeaderHarnessAndInit(t, ctx)
+
+		h.expectNewBlockProposalRequestedAndSaved(1)
+		h.expectCommitSent(1, h.config.NodePublicKey())
+		h.receivedCommittedViaGossipFromSeveral(3, 0, true, true)
```

Comment:
it's very hard to understand the meaning of these 4 arguments from reading the code. Maybe pass an argument struct so we can name the parameters?

Reply:
or even split to multiple functions, or a builder